src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -292,7 +313,8 @@ public BigInteger(ReadOnlySpan<byte> value)
                 _sign = isNegative ? unchecked((int)0xffffffff) : 0;
                 for (int i = byteCount - 1; i >= 0; i--)
                 {
-                    _sign = (_sign << 8) | value[i];
+                    int idx = isBigEndian ? byteCount - 1 - i : i;
+                    _sign = (_sign << 8) | value[idx];
```

Comment:
This is adding a branch per iteration of the loop.  You could instead branch once based on isBigEndian and have two different loops.  Don't know if it matters.

Reply:
This set, specifically, I went back and forth with across many perf runs, and it wasn't distinguishable.  I hadn't A/B tested the bigger version for moving the branch out of the loop, but should.